src/ensembl/src/content/app/entity-viewer/shared/rest/rest-adaptors/rest-gene-adaptor.ts

```diff
@@ -1,30 +1,18 @@
-import { buildTranscript } from './rest-transcript-adaptor';
+import { buildTranscriptFromLookup } from './rest-transcript-adaptor';
 
-import {
-  GeneInResponse,
-  FeatureWithParent,
-  TranscriptInResponse,
-  FeatureInResponse
-} from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/transcriptData';
-import { Strand } from 'src/content/app/entity-viewer/types/strand';
+import { GeneInResponse } from '../rest-data-fetchers/geneData';
 import { Gene } from 'src/content/app/entity-viewer/types/gene';
+import { Strand } from 'src/content/app/entity-viewer/types/strand';
 
-// transform ensembl rest /overlap data into a gene data structure
-export const restGeneAdaptor = (
-  geneId: string,
-  data: FeatureInResponse[]
-): Gene => {
-  const gene = data.find((feature) => feature.id === geneId) as GeneInResponse;
-  const transcripts = data
-    .filter((feature) => (feature as FeatureWithParent).Parent === geneId)
-    .map((transcript) =>
-      buildTranscript(transcript as TranscriptInResponse, data)
-    );
+export const restGeneAdaptor = (gene: GeneInResponse): Gene => {
+  const transcripts = gene.Transcript.filter(
```

Comment:
You are removing all non-protein-coding transcripts from the gene. Why? You shouldn't need to do this.

Reply:
Argh that would've come from the code I took from the transcript adaptor. I'll fix it.

Reply:
:heavy_check_mark